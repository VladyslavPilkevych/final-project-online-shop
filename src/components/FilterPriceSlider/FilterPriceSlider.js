/* eslint react/no-multi-comp: 0, no-console: 0 */
import React, { useState } from 'react';
import Slider from 'rc-slider';
import './FilterPriceSlider.scss';
// import '../../assets/index.less';

function FilterPriceSlider() {
  const [min, setMin] = useState(null);
  const [max, setMax] = useState(null);
  function changePriceFn(value) {
    console.log(value);
    setMin(value[0]);
    setMax(value[1]);
  }
  return (
    <div>
      <div>
        {min && max
          && (
            <div style={{ display: 'flex', width: '100%', justifyContent: 'space-evenly' }}>
              <p>
                $
                {min}
              </p>
              <p>-</p>
              <p>
                $
                {max}
              </p>
            </div>
          )}
        {/* eslint-disable react/jsx-no-bind */}
        <Slider
          range
          allowCross={false}
          min={1}
          max={100000}
          defaultValue={[1, 100000]}
          onChange={changePriceFn}
        />
        {/* eslint-eneble react/jsx-no-bind */}
      </div>
    </div>
  );
}

export default FilterPriceSlider;