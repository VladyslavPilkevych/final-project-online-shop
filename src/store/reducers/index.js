import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
