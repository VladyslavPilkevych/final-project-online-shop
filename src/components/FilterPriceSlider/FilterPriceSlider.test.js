import { render } from '@testing-library/react';
import FilterPriceSlider from './FilterPriceSlider';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <FilterPriceSlider />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot FilterPriceSlider should render', () => {
  test('Snapshot FilterPriceSlider is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});