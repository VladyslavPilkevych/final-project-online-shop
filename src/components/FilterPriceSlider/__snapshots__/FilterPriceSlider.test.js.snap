// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot FilterPriceSlider should render Snapshot FilterPriceSlider is rendered 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        style="display: flex; width: 100%; justify-content: space-evenly;"
      >
        <p>
          $
        </p>
        <p>
          -
        </p>
        <p>
          $
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
